stack/stack.py

```diff
@@ -1,25 +1,59 @@
 """
 A stack is a data structure whose primary purpose is to store and
-return elements in Last In First Out order. 
+return elements in Last In, First-Out (LIFO) order/abstract data type (the last value added to the stack will be the first value removed from the stack). It is used for recursive parsing in Natural Language Processing.
 
-1. Implement the Stack class using an array as the underlying storage structure.
+1. Implement the Stack class using a dynamic array (when elements are added or removed, the storage area is resized every time accordingly)/list as the underlying storage structure.
    Make sure the Stack tests pass.
 2. Re-implement the Stack class, this time using the linked list implementation
    as the underlying storage structure.
    Make sure the Stack tests pass.
 3. What is the difference between using an array vs. a linked list when 
    implementing a Stack?
 """
-class Stack:
+# import sys
+# sys.path.append('../singly_linked_list')
+from singly_linked_list import LinkedList
+
+class ArrayStack:
+    # Constructor
+    def __init__(self):
+        self.storage = [] # empty array - will fill up with items that are pushed to the stack
+
+    # find length of a Linked List iteratively
+    def __len__(self):
```

Comment:
Nice use of dunder-methods (something to google perhaps), they have a very useful purpose in Python.